node/lib/cmd/cherrypick.js

```diff
@@ -83,7 +83,8 @@ exports.executeableSubcommand = co.wrap(function *(args) {
     const Status     = require("../util/status");
 
     const repo = yield GitUtil.getCurrentRepo();
-    yield Status.ensureCleanAndConsistent(repo);
+    const status = yield Status.getRepoStatus(repo);
+    Status.ensureCleanAndConsistent(status);
```

Comment:
I'm not actually refactoring this part of the code, but I would probably not want choose to factor it that way if I did. It may be too philosophical to discuss in a review, but `RepoStatus` is what I call a "value" type -- it has almost no behavior associated with it; it's job is just to represent the status of a repository. Business logic, such as deciding what "consistent" means is something I put into separate modules.